src/main.py

```diff
@@ -93,44 +97,101 @@ def main() -> None:
                 "Can't exclude a goal predicate!"
     else:
         preds = env.predicates
+    # Create the agent (approach).
     approach = create_approach(CFG.approach, preds, env.options, env.types,
                                env.action_space)
-    # If approach is learning-based, get training datasets and do learning,
-    # testing after each learning call. Otherwise, just do testing.
+    # If agent is learning-based, generate offline dataset, allow the agent
+    # to learn from it, and then proceed with the online learning loop. Test
+    # after each learning call. If agent is not learning-based, just test once.
     if approach.is_learning_based:
-        if CFG.load_approach:
+        train_tasks = env.get_train_tasks()
+        dataset, total_num_transitions = _generate_or_load_offline_dataset(
```

Comment:
is `total_num_transitions` not something that we can just easily get from `dataset`?